pkg/syslLogger/api_test.go

```diff
@@ -0,0 +1,239 @@
+			return func(tt *testing.T) {
+				tt.Parallel()
+
+				logger := loggers.NewMockLogger()
+				logger.On("WithField", sc.Key, sc.Val).Return(logger)
+				logger.On("Copy").Return(loggers.NewMockLogger())
+				ctx := context.WithValue(context.Background(), loggerKey, logger)
+
+				newCtx := AddField(ctx, sc.Key, sc.Val)
+				newLogger := newCtx.Value(loggerKey).(*loggers.MockLogger)
+
+				require.True(tt, logger.AssertExpectations(tt))
+				require.True(tt, &ctx != &newCtx)
+				require.True(tt, &logger != &newLogger)
+			}
+		}(c))
+	}
+}
+
+func TestLogField(t *testing.T) {
+	cases := testutil.GenerateSingleFieldCases()
+	for _, c := range cases {
+		t.Run(c.Name, func(sc testutil.SingleField) func(*testing.T) {
+			return func(tt *testing.T) {
+				tt.Parallel()
+
+				logger := loggers.NewMockLogger()
+
+				// this is done so that the mock logger is of the
+				// same reference, for testing purposes
+				logger.On("Copy").Return(logger)
+				logger.On("WithField", sc.Key, sc.Val).Return(loggers.NewMockLogger())
+				ctx := context.WithValue(context.Background(), loggerKey, logger)
+
+				newLogger := LogField(ctx, sc.Key, sc.Val).(*loggers.MockLogger)
+
+				require.True(tt, logger.AssertExpectations(tt))
+				require.True(tt, &logger != &newLogger)
+			}
+		}(c))
+	}
+}
+
+func TestAddFields(t *testing.T) {
+	cases := testutil.GenerateMultipleFieldCases()
+	for _, c := range cases {
+		t.Run(c.Name, func(mc testutil.MultipleFields) func(*testing.T) {
+			return func(tt *testing.T) {
+				tt.Parallel()
+
+				logger := loggers.NewMockLogger()
+				logger.On("WithFields", mc.Fields).Return(logger)
+				logger.On("Copy").Return(loggers.NewMockLogger())
+				ctx := context.WithValue(context.Background(), loggerKey, logger)
+
+				newCtx := AddFields(ctx, mc.Fields)
+				newLogger := newCtx.Value(loggerKey).(*loggers.MockLogger)
+
+				require.True(tt, logger.AssertExpectations(tt))
+				require.True(tt, &ctx != &newCtx)
+				require.True(tt, &logger != &newLogger)
+			}
+		}(c))
+	}
+}
+
+func TestLogFields(t *testing.T) {
+	cases := testutil.GenerateMultipleFieldCases()
+	for _, c := range cases {
+		t.Run(c.Name, func(mc testutil.MultipleFields) func(*testing.T) {
+			return func(tt *testing.T) {
+				tt.Parallel()
+
+				logger := loggers.NewMockLogger()
+
+				// this is done so that the mock logger is of the
+				// same reference, for testing purposes
+				logger.On("Copy").Return(logger)
+				logger.On("WithFields", mc.Fields).Return(loggers.NewMockLogger())
+				ctx := context.WithValue(context.Background(), loggerKey, logger)
+
+				newLogger := LogFields(ctx, mc.Fields).(*loggers.MockLogger)
+
+				require.True(tt, logger.AssertExpectations(tt))
+				require.True(tt, &logger != &newLogger)
+			}
+		}(c))
+	}
+}
+
+func testAllCases(t *testing.T, name string, test func(testutil.LogCases, *testing.T) func(*testing.T)) {
+	for _, c := range testutil.GenerateLogCases() {
+		t.Run(name+" "+c.Name, test(c, t))
+	}
+}
```

Comment:
Simpler:
```go
func testAllCases(t *testing.T, name string, test func(testutil.LogCases, *testing.T)) {
	for _, c := range testutil.GenerateLogCases() {
		t.Run(name+" "+c.Name, func(tt *testing.T) { test(c, tt) })
	}
}
```

Reply:
it was done that way to avoid `scopelint` issues

Reply:
fixed